### ISSUE TITLE
[Feature]: Use min/max metric value over all epochs on sweep

### ISSUE THREAD

**Issue by FredHaa** [wandb/wandb#3468]
### Description

When running a sweep the value of each runs is the value of the metric after the run has completed.
However, in some cases (e.g. when training on synthetic data and evaluating on real data) the metric varies a lot over epochs, and therefore the last value is often not the best.

It would therefore be nice if you could set the metric goal to e.g. 'global_maximum' and 'global_minimum' such that a set of hyperparameters is evaluated based on the best metric reached throughout the run and not just the final value.

### Suggested Solution

When setting the metric goal to global_maximum the maximum value over all logged metrics is returned for the run.
Same with global_minimum

**Comment by FredHaa** [comment]
Closed issue as the solution is obviously to just keep track of the maximum/minimum values in the code, and then log those values in the end of the run.

**Comment by nick-maykr** [comment]
Actually, this would be pretty useful. For example, if we use the huggingface integration with wandb through the `trainer.hyperparameter_search()` function, there's currently **no way** to log the min/max metric value observed throughout all the epochs.
I feel like the change at wandb side would be pretty minor - just to add the min/max flag to the value dropdown on graphs. 
@FredHaa what do you think of reopening?)

**Comment by Wocck** [comment]
Hi, we are team of students and we are intrested in solving this issue. However this is our first encounter with such a big project as Wandb and we don't know where to start and search for the source code responsible for this functionality. Could you maybe point us in the right direction so it would be easier for us to start our work ;)

**Comment by V0XNIHILI** [comment]
Totally agree, this would be super useful!

**Comment by gamemeine** [comment]
Hi everyone,

We are starting work on implementing this feature as our team project. More updates will follow.
